src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathFilterTest.java

```diff
@@ -179,4 +180,64 @@ private static SuppressionXpathFilter createSuppressionXpathFilter(String fileNa
         return suppressionXpathFilter;
     }
 
+    @Test
+    public void testFalseEncodeString() throws Exception {
+        final boolean optional = false;
+        final SuppressionXpathFilter filter = createSuppressionXpathFilter(
+                getPath("InputSuppressionXpathFilterEscapeString.xml"), optional);
+        final File file = new File(getPath("InputSuppressionXpathFilterEscapeString.java"));
+
+        assertFalse(filter.accept(createTreeWalkerAudit(8, 37, TokenTypes.PLUS, file)),
+                "TreeWalker audit event should be rejected");
```

Comment:
It would be great. Just a small issue, since the approach we take, will regression test is we check a file which only has a single violation, and a single Xpath is given for that suppression. That approach would lead to a lot file, with each file containing a single violation. So, how would we escape that. Please explain.

Reply:
>  will regression test is we check a file which only has a single violation, and a single Xpath is given for that suppression

it could contain one of few. Execution without suppression files confirms amount of violations, execution with suppression file result in no violations.
Example - https://github.com/checkstyle/checkstyle/blob/4860e1005ea970fa5d177cc2640b1f34a3a92a96/src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionMatchXpathTest.java#L42
please run this test in debug to see it in action.

Reply:
Sir, will I was trying to implement this test, what I found was these tests extends to `AbstractXpathTestSupport` and our normal tests and supported by `AbstractPathTestSupport` and `AbstractModuleTestSupport` which don't have funtionality to process xpath and verify. 
The `AbstractXpathTestSupport` has root directory of `src/it` but we cannot use that. So, either we have add those functionality to our other interfaces or change the directory of `AbstractXpathTestSupport` and both of them sounds good to be.
Also, only one of them can be applied at a time (Keeping the class as `class` and not `interface`) so the entire test have to be changed to newer one. 
Please let me know the solutions to this problems or correct me if I am wrong in any of the above. Thanks.

Reply:
@romani ping.

Reply:
ok, lets keep it as is for now, we will resolve this scope of bigger project.

Reply:
Sir, should I move ahead and create a new Issue for this, and then work on it to create the basic framework for this. I really like this idea and would like to work on it. Except semaphorica all other test were passed, and if I give some more time, I think I can really implement it, just some I require your and other members help a bit.